packages/api/amplify_api/lib/src/api_plugin_impl_flutter.dart

```diff
@@ -0,0 +1,119 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'dart:io';
+
+import 'package:amplify_api/src/api_plugin_impl.dart';
+import 'package:amplify_api/src/native_api_plugin.dart';
+import 'package:amplify_core/amplify_core.dart';
+import 'package:connectivity_plus/connectivity_plus.dart';
+import 'package:flutter/services.dart';
+
+/// {@template amplify_api.amplify_api}
+/// The AWS implementation of the Amplify API category for Flutter.
+/// {@endtemplate}
+class AmplifyAPI extends AmplifyAPIDart with AWSDebuggable {
+  /// {@macro amplify_api.amplify_api}
+  AmplifyAPI({
+    List<APIAuthProvider> authProviders = const [],
+    super.baseHttpClient,
+    super.modelProvider,
+    super.subscriptionOptions,
+  }) : super(
+          authProviders: authProviders,
+          networkStreamGenerator: () => Connectivity()
+              .onConnectivityChanged
+              .map((ConnectivityResult connectivityResult) {
+            switch (connectivityResult) {
+              case ConnectivityResult.none:
+                return false;
+              default:
+                return true;
```

Comment:
Callout, there were 2 enums not being used before: VPN and bluetooth. I figured simpler to do this way, although one could argue they might introduce some value in the future that should map to `false` so I could also be convinced to list them out again.